src/components/Popup/Popup.tsx

```diff
@@ -96,56 +109,97 @@ export default class Popup extends UIComponent<Extendable<IPopupProps>, IPopupSt
     as: Portal,
     align: 'start',
     position: 'above',
+    accessibility: PopupBehavior as Accessibility,
   }
 
-  public state = { triggerRef: undefined }
+  public state = { triggerRef: undefined, popupOpened: false }
+
+  actionHandlers: AccessibilityActionHandlers = {
+    performClick: e => this.onTriggerClick(e),
+    closeAndFocusTrigger: e => this.handlePopupState(e, false, this.focusTrigger),
+  }
 
   public renderComponent({
     ElementType,
     classes,
+    accessibility,
     rest,
     rtl,
   }: IRenderResultConfig<IPopupProps>): React.ReactNode {
-    const { children, trigger } = this.props
+    const { children, trigger, position, align } = this.props
 
     return (
       <ElementType
         className={classes.root}
         {...rest}
+        open={this.state.popupOpened}
         trigger={trigger}
         triggerRef={this.handleTriggerRef}
+        triggerAccessibility={{
+          ...accessibility.attributes.trigger,
```

Comment:
won't it be better to use the object of the following structure instead, so that attributes and keyHandlers will be explicitly splitted? Seems to be the safest option at the first stages - and it will resemble we currently have for accessibility behavior objects 

```jsx
triggerAccessibility={{
  attributes: accessibility.attributes.trigger,
  handlers: accessibility.keyHandlers.trigger
}}
```

Reply:
It always good to explicitly set values. But then in ```Portal``` will need to handle it as object
```
{
   attributes:...
   handlers:...
}
```

Which brings us to handling one more Type.

Currently, we are safely passing from ```Popup``` trigger related attributes
And in ```Portal``` we expect ```triggerAccessibility?: IAccessibilityAttributes & OnKeyDownHandler``` which also applies some constraints and make it safer

What do you think?

Reply:
sorry, but still not sure that this is the reason to avoild explicitness, as the new style is introduced even now by using `IAccessibilityAttributes & OnKeyDownHandler` expression - and there is no enforcement for this type on the side where object is provided. The same could be achieved if type literal object will be used instead:

```js
triggerAccessibility?: { attributes: IAccessibilityAttributes, handlers: OnKeyDownHandler }
```

Lets keep this in mind and try to address this thing in future - especially given that we both agree on the benefits of having these two being provided as explicit props

Reply:
actually, I understood you a bit wrong from the 1st message, sorry. Agree with this proposal and will make the change. Thank you!